File: dsa-python/generatePermutations.py
# Generate permutations using recursion
def Generate(permutation, elements, positions):
    
   if( len(permutation) == len(elements) ):

       for it in permutation:
          print(it, end=' ')
       print('\n')

   else:

       for i in range (0, len(elements)):

           if (positions[i]):
               continue

           # Set the position (taken), append the element
           positions[i] = True
           permutation.append(elements[i])

           Generate (permutation, elements, positions)
 
           # Remove the element, reset the position (available), 
           permutation.pop()
           positions[i] = False
           

elements = []
permutation = []
size = input("Size : ")
for i in range(0, int(size)):
   x = input("Element " + str(i) + " : ")
   elements.append(x)

positions = [False] * int(size)

print("\nPermutations\n")
Generate(permutation, elements, positions)

File: dsa-javascript/generatePermutations.js
function generatePermutations(permutation, elements, positions) {
    if(permutation.length === elements.length) {
        console.log(permutation);
    } else {
        for(let i = 0; i < elements.length; i++) {
            if(positions[i]) continue;

            // set pos --> true; push element to permutation array
            positions[i] = true;
            permutation.push(elements[i]);

            // recursive call to get different permutations for current perm, pos array
            generatePermutations(permutation, elements, positions);

            // remove element from permutation array; set pos --> false
            permutation.pop();
            positions[i] = false;
        }
    }
}

let elements = [1,2,3];
permutation = [];
let size = 3;
// fill an array with false 
positions = new Array(size).fill(false);

console.log("\npermutations\n");
generatePermutations(permutation,elements,positions);